pkg/configurator/validating_webhook.go

```diff
@@ -330,17 +332,22 @@ func reasonForDenial(resp *v1beta1.AdmissionResponse, mustBe string, field strin
 }
 
 // getPartialValidatingWebhookConfiguration returns only the portion of the ValidatingWebhookConfiguration that needs to be updated.
-func getPartialValidatingWebhookConfiguration(webhookName string, cert certificate.Certificater, webhookConfigName string) admissionv1beta1.ValidatingWebhookConfiguration {
-	return admissionv1beta1.ValidatingWebhookConfiguration{
+func getPartialValidatingWebhookConfiguration(webhookName string, cert certificate.Certificater, webhookConfigName string) admissionregv1.ValidatingWebhookConfiguration {
+	return admissionregv1.ValidatingWebhookConfiguration{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: webhookConfigName,
 		},
-		Webhooks: []admissionv1beta1.ValidatingWebhook{
+		Webhooks: []admissionregv1.ValidatingWebhook{
 			{
 				Name: webhookName,
-				ClientConfig: admissionv1beta1.WebhookClientConfig{
+				ClientConfig: admissionregv1.WebhookClientConfig{
 					CABundle: cert.GetCertificateChain(),
 				},
+				SideEffects: func() *admissionregv1.SideEffectClass {
+					sideEffect := admissionregv1.SideEffectClassNone
+					return &sideEffect
+				}(),
+				AdmissionReviewVersions: []string{"v1"},
```

Comment:
If these fields were added to the chart template and this is only used to create a patch, is it necessary to define them here?

Reply:
This must be specified, else the Patch() will fail with the following error:
```{"level":"fatal","component":"kube-events","error":"ValidatingWebhookConfiguration.admissionregistration.k8s.io \"osm-webhook-osm\" is invalid: webhooks[0].admissionReviewVersions: Required value: must specify one of v1, v1beta1","reason":"FatalInitializationError","time":"2021-03-29T20:51:22Z","file":"events.go:138","message":"Error creating osm-config validating webhook"}
```